Essentials/src/net/ess3/api/events/MuteStatusChangeEvent.java

```diff
@@ -4,7 +4,35 @@
 
 
 public class MuteStatusChangeEvent extends StatusChangeEvent {
+    private Long timestamp;
+    private String reason;
+
+    @Deprecated
     public MuteStatusChangeEvent(IUser affected, IUser controller, boolean value) {
+        this(affected, controller, value, null, null);
+    }
+
+    public MuteStatusChangeEvent(IUser affected, IUser controller, boolean value, Long timestamp, String reason) {
         super(affected, controller, value);
+        this.timestamp = timestamp;
+        if (reason == null) {
+            this.reason = null;
+        } else {
+            this.reason = reason.isEmpty() ? null : reason;
+        }
+    }
+
+    /**
+     * @return Returns null if the timestamp is unknown due to usage of deprecated api; Otherwise, the timestamp with 0 indicating permanent.
```

Comment:
The second portion of this description reads poorly in my opinion. `Otherwise, the timestamp with 0` should probably be `Otherwise, the timestamp, or 0` to make sure it's clear.

Reply:
👍